checker/src/abstract_value.rs

```diff
@@ -307,6 +295,25 @@ impl AbstractValue {
     pub fn make_typed_unknown(var_type: ExpressionType, path: Rc<Path>) -> Rc<AbstractValue> {
         AbstractValue::make_from(Expression::Variable { path, var_type }, 1)
     }
+
+    /// Creates an abstract value that is basically the same as the given value, except that
+    /// a tag domain element is attached to it. This routine is useful when a value is being
+    /// refined and we don't want to drop the existing tags associated with that value.
+    #[logfn_inputs(TRACE)]
+    pub fn make_with_tags(
+        value: &Rc<AbstractValue>,
+        tags_opt: Option<&Rc<TagDomain>>,
+    ) -> Rc<AbstractValue> {
+        match tags_opt {
+            None => value.clone(),
+            Some(tags) => Rc::new(AbstractValue {
+                expression: value.expression.clone(),
```

Comment:
As discussed earlier, this will change to a new expression tag in a future PR.

Reply:
Added a todo comment.